airbyte-webapp/src/components/CreateConnection/CreateConnectionForm.tsx

```diff
@@ -0,0 +1,159 @@
+import { Form, Formik, FormikHelpers } from "formik";
+import React, { Suspense, useCallback } from "react";
+import { useNavigate } from "react-router-dom";
+import { useToggle } from "react-use";
+
+import { FormChangeTracker } from "components/FormChangeTracker";
+import LoadingSchema from "components/LoadingSchema";
+
+import { DestinationRead, SourceRead } from "core/request/AirbyteClient";
+import {
+  ConnectionFormServiceProvider,
+  tidyConnectionFormValues,
+  useConnectionFormService,
+} from "hooks/services/ConnectionForm/ConnectionFormService";
+import { useFormChangeTrackerService, useUniqueFormId } from "hooks/services/FormChangeTracker";
+import { useCreateConnection } from "hooks/services/useConnectionHook";
+import { SchemaError as SchemaErrorType, useDiscoverSchema } from "hooks/services/useSourceHook";
+import { useCurrentWorkspaceId } from "services/workspaces/WorkspacesService";
+import CreateControls from "views/Connection/ConnectionForm/components/CreateControls";
+import { OperationsSection } from "views/Connection/ConnectionForm/components/OperationsSection";
+import { ConnectionFormFields } from "views/Connection/ConnectionForm/ConnectionFormFields";
+import { connectionValidationSchema, FormikConnectionFormValues } from "views/Connection/ConnectionForm/formConfig";
+
+import styles from "./CreateConnectionForm.module.scss";
+import { CreateConnectionNameField } from "./CreateConnectionNameField";
+import { SchemaError } from "./SchemaError";
+
+interface CreateConnectionProps {
+  source: SourceRead;
+  destination: DestinationRead;
+  afterSubmitConnection?: () => void;
+}
+
+interface CreateConnectionPropsInner extends Pick<CreateConnectionProps, "afterSubmitConnection"> {
+  schemaError: SchemaErrorType;
+}
+
+const CreateConnectionFormInner: React.FC<CreateConnectionPropsInner> = ({ schemaError, afterSubmitConnection }) => {
+  const navigate = useNavigate();
+
+  const { mutateAsync: createConnection } = useCreateConnection();
+
+  const { clearFormChange } = useFormChangeTrackerService();
+
+  const workspaceId = useCurrentWorkspaceId();
+  const formId = useUniqueFormId();
+
+  const { connection, initialValues, mode, getErrorMessage, setSubmitError } = useConnectionFormService();
+  const [editingTransformation, toggleEditingTransformation] = useToggle(false);
```

Comment:
Is there a specific reason you used a toggle here instead of a normal boolean state? I think either will probably work fine, but I remember seeing some other weird edge case issue with toggles before that made me feel like using a normal boolean state and explicitly setting it to `true` or `false` in the relevant code paths was more reliable